### ISSUE TITLE
Make build failed

### ISSUE THREAD

**Issue by concefly** [segmentio/analytics.js#511]
Hi.

I tried to build a custom distribution of `analytics.js` with the **master branch**, but I got something wrong.

```
$ make build

     building : lib/index.js
        using : compatibility
        using : stoj
        error : TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at extname (path.js:1429:5)
    at extension (/Users/huize/project/my/analytics.js/node_modules/duo/node_modules/duo-main/index.js:135:10)
    at dependency (/Users/huize/project/my/analytics.js/node_modules/duo/node_modules/duo-main/index.js:83:16)
    at coerce (/Users/huize/project/my/analytics.js/node_modules/duo/node_modules/duo-main/index.js:31:7)
    at Duo.resolve (/Users/huize/project/my/analytics.js/node_modules/duo/lib/duo.js:798:22)
    at next (native)
    at next (/Users/huize/project/my/analytics.js/node_modules/duo/node_modules/co/index.js:74:21)
    at Duo.<anonymous> (/Users/huize/project/my/analytics.js/node_modules/duo/node_modules/co/index.js:45:5)
    at Duo.next (/Users/huize/project/my/analytics.js/node_modules/duo/node_modules/co/index.js:90:21)
    at Duo.<anonymous> (/Users/huize/project/my/analytics.js/node_modules/duo/node_modules/co/index.js:93:18)
    at Immediate._onImmediate (/Users/huize/project/my/analytics.js/node_modules/duo/node_modules/co/index.js:52:14)
    at tryOnImmediate (timers.js:543:15)
    at processImmediate [as _immediateCallback] (timers.js:523:5)
```

Could you please tell me what was wrong?

NOTE:

- `nodejs`==v6.2
- `npm`==v3.8.9

**Comment by concefly** [comment]
I change this function in `node_modules/duo/lib/duo.js` then it works successfully.
To be honest I don't know why ...

```javascript
Duo.prototype.resolve = function *(dep, file, entry) {
  var isManifest = this.manifest() === basename(dep);
  var ext = extension(dep) ? '' : '.' + entry.type;
  var path = resolve(dirname(file.path), dep);
  var isRelative = dep.slice(0, 2) === './';
  var isParent = dep.slice(0, 2) === '..';
  var isAbsolute = dep[0] === '/';
  var ret;

  if (isManifest) {
    // `dep` is a component.json
    var json = readJson(resolve(file.root, dep));
    // var entrypoint = main(json, entry.type) || 'index.' + entry.type;
    var entrypoint = 'index.' + entry.type;
    ret = resolve(file.root, dirname(dep), entrypoint);
  } else if (isAbsolute) {
    // `dep` is an absolute path (relative to app or component root)
    var relroot = this.findRoot(file.path);
    ret = join(relroot, dep.replace(relroot, ''));
  } else if (isParent) {
    // `dep` is a parent path with ".."
    ret = path;
  } else if (isRelative) {
    // `dep` is a relative path with "./"
    ret = path;
  } else if (yield isRelativeCSS(entry, path)) {
    // Hack to support for CSS relative paths without "./"
    // Example: body { background-image: url('image.jpg'); }
    ret = path;
  }

  // Could not resolve
  if (!ret) return;

  // Strip any hashes, querystrings, etc from path.
  // fonts/glyphicons-halflings-regular.eot?#iefix
  ret = stripPath(ret);

  // does one of these file paths exist?
  //
  // ./file.{{ext}}
  // ./file/index.{{ext}}
  return yield exists([
    ret + ext,
    join(ret, 'index' + ext)
  ]);

  // check filesystem for relative asset
  function *isRelativeCSS(entry, path) {
    if (entry.type !== 'css') return false;
    path = stripPath(path);
    if (!extension(path)) return false;
    return yield exists(path);
  }
};
```

**Comment by elfeffe** [comment]
I have the same problem, what did you change?

**Comment by elfeffe** [comment]
Ok, this
`    // var entrypoint = main(json, entry.type) || 'index.' + entry.type;`

**Comment by elfeffe** [comment]
the problem with your solution is that it works, but makes an empty file. not empty, but does not include the integrations.
I posted correct fix here https://github.com/segmentio/analytics.js/issues/512

**Comment by concefly** [comment]
@elfeffe Well, thanks

**Comment by OskJonsdottir** [comment]
This is due to a bug in duo-main, fixed here: https://github.com/duojs/main/commit/f40e8fe70ca7efac5185c96bb2aa069816c14b15

That fix was released in version 0.15.7 of duo, but analytics.js uses ^0.12.0
